internal/v1/handler_blueprints_test.go

```diff
@@ -1696,3 +1696,238 @@ func TestFixupBlueprint(t *testing.T) {
 		require.NoError(t, srv.DB.DeleteBlueprint(context.Background(), bpID, "000000", "000000"))
 	}
 }
+
+func TestBlueprintComplianceSnapshot(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("crypt() not supported on darwin")
+	}
+
+	ctx := context.Background()
+	dbase, srvURL, shutdownFn := makeTestServer(t, nil)
+	defer shutdownFn(t)
+
+	body := map[string]interface{}{
+		"name":        "BlueprintWithCompliance",
+		"description": "Blueprint with compliance policy",
+		"customizations": map[string]interface{}{
+			"packages": []string{"nginx"},
+			"openscap": map[string]interface{}{
+				"policy_id": mocks.PolicyID,
+			},
+		},
+		"distribution": "rhel-8",
+		"image_requests": []map[string]interface{}{
+			{
+				"architecture":   "x86_64",
+				"image_type":     "aws",
+				"upload_request": map[string]interface{}{"type": "aws", "options": map[string]interface{}{"share_with_accounts": []string{"test-account"}}},
+			},
+		},
+	}
+
+	statusCode, respBody := tutils.PostResponseBody(t, srvURL+"/api/image-builder/v1/blueprints", body)
+	require.Equal(t, http.StatusCreated, statusCode)
+
+	var result v1.CreateBlueprintResponse
+	err := json.Unmarshal([]byte(respBody), &result)
+	require.NoError(t, err)
+
+	blueprintEntry, err := dbase.GetBlueprint(ctx, result.Id, "000000", nil)
+	require.NoError(t, err)
+	require.NotNil(t, blueprintEntry.ServiceSnapshots)
+
+	var serviceSnapshots db.ServiceSnapshots
+	err = json.Unmarshal(blueprintEntry.ServiceSnapshots, &serviceSnapshots)
+	require.NoError(t, err)
+	require.NotNil(t, serviceSnapshots.Compliance)
+	require.Equal(t, mocks.PolicyID, serviceSnapshots.Compliance.PolicyId.String())
+	require.NotEmpty(t, serviceSnapshots.Compliance.PolicyCustomizations)
+
+	var policyCustomizations map[string]interface{}
+	err = json.Unmarshal(serviceSnapshots.Compliance.PolicyCustomizations, &policyCustomizations)
+	require.NoError(t, err, "PolicyCustomizations should be valid JSON")
+	require.NotEmpty(t, policyCustomizations)
+
+	require.NoError(t, dbase.DeleteBlueprint(ctx, result.Id, "000000", "000000"))
+}
+
+func TestBlueprintCreationRollbackOnPolicyFailure(t *testing.T) {
+	ctx := context.Background()
+	dbase, srvURL, shutdownFn := makeTestServer(t, nil)
+	defer shutdownFn(t)
+
+	body := map[string]interface{}{
+		"name":        "BlueprintShouldRollback",
+		"description": "Blueprint that should be rolled back",
+		"customizations": map[string]interface{}{
+			"packages": []string{"nginx"},
```

Comment:
Let's keep the blueprint minimal here, with only the openscap customization set.